### DISCUSSION TITLE
The shape of mean and the standard deviation must be identical

### DISCUSSION THREAD

**Discussion by estheroratemstet** [Trusted-AI/adversarial-robustness-toolbox#1760] [Q&A] [answered]
I have a problem for 
**patch, patch_mask = ap.generate(x=images, y=y_target)**
The ValueError: The shape of the mean and the standard deviation must be identical.
What do I need to do ??

**Comment by estheroratemstet** [comment, marked as answer]
**I was able to solve the problem**
need to change the preprocessing instead <br>
**preprocessing=([mean_b, mean_g, mean_r], 1)** <br>
  to <br>
**preprocessing=(np.array([mean_b, mean_g, mean_r]), np.array([1.0, 1.0, 1.0]))** <br>

**instead write this code**

tfc = TensorFlowClassifier(clip_values=clip_values, input_ph=_image_input, labels_ph=_target_ys,
                           output=_logits, sess=sess, loss=target_loss,
                           preprocessing=([mean_b, mean_g, mean_r], 1))

**write this code**


tfc = TensorFlowClassifier(clip_values=clip_values, input_ph=_image_input, labels_ph=_target_ys,
                           output=_logits, sess=sess, loss=target_loss,
                           preprocessing=(np.array([mean_b, mean_g, mean_r]), np.array([1.0, 1.0, 1.0]))
                           )

**Reply by beat-buesser** [reply]
Hi @estheroratemstet That's great! Thank you for using ART!
